### PR TITLE
feat(provider/ecs): ECS Target Group cache client and tests.

### PR DESCRIPTION
**Pull request by dkirillov**
Adds an EcsTargetGroupCacheClient which  allows providers/services to interface more easily with the already existing AWS target group cache.

@cfieber @ajordens @robzienert @BrunoCarrier

### CODE REVIEW

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/EcsTargetGroupCacheClient.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+    Set<Map<String, Object>> targetGroupAttributes = fetchLoadBalancerAttributes(targetGroupKeys);
+
+    List<EcsTargetGroup> targetGroups = convertToTargetGroup(targetGroupAttributes);
+
+    return targetGroups;
+  }
+
+  private EcsTargetGroup convertToTargetGroup(Map<String, Object> targetGroupAttributes) {
+    targetGroupAttributes.toString();
```
Comment (robzienert) [inline clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/EcsTargetGroupCacheClient.java]: Should remove this line.

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/EcsTargetGroupCacheClient.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+    Set<CacheData> targetGroupsWithAtLeastOneLoadBalancer = targetGroups
+      .stream()
+      .filter(targetGroup -> targetGroup.getRelationships().get("loadBalancers") != null
+        && targetGroup.getRelationships().get("loadBalancers").size() > 0)
+      .collect(Collectors.toSet());
+
+    Set<Map<String, Object>> targetGroupAttributes = targetGroupsWithAtLeastOneLoadBalancer.stream()
+      .map(lb -> lb.getAttributes())
+      .collect(Collectors.toSet());
```
Comment (robzienert) [inline clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/EcsTargetGroupCacheClient.java]: Should probably just combine these two streams.

```
// ...
.filter(EcsTargetGroupCacheClient::hasLoadBalancers)
.map(lb -> lb.getAttributes)
.collect(Collectors.toSet());
```

Bonus points for moving the filter logic into its own function so it's a little easier to read (purely my opinion here).

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/EcsTargetGroupCacheClient.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  }
+
+
+  private String extractAccount(EcsTargetGroup ecsTargetGroup) {
+    return null; // Implement this based on ecsTargetGroup.getId(); -->  example value: aws:targetGroups:continuous-delivery:us-west-2:clouddriver-dev:vpc-bb8f37dd
+  }
+
+  private String extractRegion(EcsTargetGroup ecsTargetGroup) {
+    return null; // Implement this based on ecsTargetGroup.getId(); -->  example value: aws:targetGroups:continuous-delivery:us-west-2:clouddriver-dev:vpc-bb8f37dd
+  }
```
Comment (robzienert) [inline clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/EcsTargetGroupCacheClient.java]: Do these two methods need to be implemented before the class will work?

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/EcsTargetGroupCacheClient.java (hunk 1 above)
Comment (dkirillov) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/EcsTargetGroupCacheClient.java]: Done.

Thanks!

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/EcsTargetGroupCacheClient.java (hunk 2 above)
Comment (dkirillov) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/EcsTargetGroupCacheClient.java]: Agreed, it does look better/easier to read.

Done.

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/EcsTargetGroupCacheClient.java (hunk 3 above)
Comment (dkirillov) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/EcsTargetGroupCacheClient.java]: Nope - removed.

**Pull request merged**